packages/astro/src/i18n/index.ts

```diff
@@ -19,6 +19,17 @@ export function requestHasLocale(locales: Locales) {
 	};
 }
 
+export function requestIs404Or500(request: Request, base?: string) {
+	const url = new URL(request.url);
+
+	return (
+		url.pathname.startsWith('/404') ||
+		url.pathname.startsWith(base + '/404') ||
```

Comment:
I think this is wrong if base is undefined? i think it will return `undefined/404`. Maybe this could simplified by making the default value of `base` an empty sting and always checking with the base

Reply:
I hate JS